File: String/1689. Partitioning Into Minimum Number Of Deci-Binary Numbers/MInPartitions.go
func minPartitions(n string) int {
	maxDigit := 0
	for _, digit := range n {
		currentDigit := int(digit - '0')
		if currentDigit > maxDigit {
			maxDigit = currentDigit
		}
		if maxDigit == 9 {
			break
		}
	}
	return maxDigit
}

File: String/1689. Partitioning Into Minimum Number Of Deci-Binary Numbers/MinPartitions.rs
impl Solution {
    pub fn min_partitions(n: String) -> i32 {
        let mut result = 0;
        for c in n.chars() {
            let current_digit = c as i32 - '0' as i32;
            if current_digit > 0 {
                if result < current_digit {
                    result = current_digit;
                }
            }
            if result == 9 {
                break;
            }
        }
        result
    }
}